### ISSUE TITLE
Is there a way to implement early stopping if we start overfitting on validation?

### ISSUE THREAD

**Issue by kanodiaayush** [coreylynch/pyFM#47]
This would greatly help with speed.

**Comment by kanodiaayush** [comment]
In fact, we could access the model with the lowest error on the validation set, this would help greatly with hyperparameter tuning.
